### PR TITLE
Use code from current version of rufus to dismiss windows format popups

### PR DESCRIPTION
**Pull request by kendase3**
TODO: work for different (non-english) locales

Addresses OVER-6235

I still have some cleaning up to do but YGtI

### CODE REVIEW

File: src/shared.h [hunk 1]
```diff
@@ -60,4 +60,8 @@ files
 
 #define safe_strdup _strdup
 
+// from bled
```
Comment (nicholasbishop) [inline src/shared.h]: What's bled?

Comment (kendase3) [reply src/shared.h]: it's a subproject of rufus

File: src/dismissprompt.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#define DISMISSPROMPT_H_
+
+#include <commdlg.h>
+#include <richedit.h>
+#include <shlobj.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <windows.h>
+#include <windowsx.h>
```
Comment (nicholasbishop) [inline src/dismissprompt.h]: This seems like a lot of includes for an almost-empty header file, can some of these go to the .cc file?

File: src/msapi_utf8.h [hunk 3]
```diff
@@ -116,16 +226,17 @@ static __inline HANDLE CreateFileU(const char* lpFileName,
...
+static __inline BOOL SetupDiGetDeviceRegistryPropertyU(
     HDEVINFO DeviceInfoSet,
     PSP_DEVINFO_DATA DeviceInfoData,
     DWORD Property,
     PDWORD PropertyRegDataType,
     PBYTE PropertyBuffer,
     DWORD PropertyBufferSize,
     PDWORD RequiredSize) {
-  bool ret = false;
+  BOOL ret = FALSE;
```
Comment (nicholasbishop) [inline src/msapi_utf8.h]: These bool changes seem unnecessary

File: src/msapi_utf8.h [hunk 4]
```diff
@@ -23,16 +23,49 @@
...
+  }
+
+// Never ever use isdigit() or isspace(), etc. on UTF-8 strings!
+// These calls take an int and char is signed so MS compilers will produce an
+// assert error on anything that's > 0x80
+#define isasciiU(c) isascii((unsigned char)(c))
+#define iscntrlU(c) iscntrl((unsigned char)(c))
+#define isdigitU(c) isdigit((unsigned char)(c))
+#define isspaceU(c) isspace((unsigned char)(c))
+#define isxdigitU(c) isxdigit((unsigned char)(c))
```
Comment (nicholasbishop) [inline src/msapi_utf8.h]: Are any of these used?

**Comment by kendase3** [comment]
closing for now; i frogot to separate out this code into the windows-specific implementation.  i'll have to think about the cleanest way to do that (i had it as part of the lib rather than the app).

**Comment by kendase3** [comment]
i'm testing this out now but it should address the windows/linux thing

**Comment by kendase3** [comment]
so i've tested this out and the only corner case i've found is that if you already had a windows format popup from before you launched gondar and you never focus on that window until you close gondar, it will still be there.  that seems like an ok corner case for now.  i'm going to work on the translation bit next.

**Comment by kendase3** [comment]
retest this please

**Comment by kendase3** [comment]
i've made separate tickets for the non-english support and the corner case mentioned above

File: src/msapi_utf8.h [hunk 5]
```diff
@@ -33,6 +34,24 @@ extern "C" {
...
+#define GetSaveFileNameU(p) GetOpenSaveFileNameU(p, TRUE)
+#define GetOpenFileNameU(p) GetOpenSaveFileNameU(p, FALSE)
+#define ListView_SetItemTextU(hwndLV, i, iSubItem_, pszText_)           \
+  {                                                                     \
+    LVITEMW _ms_wlvi;                                                   \
+    _ms_wlvi.iSubItem = iSubItem_;                                      \
+    _ms_wlvi.pszText = utf8_to_wchar(pszText_);                         \
+    SNDMSG((hwndLV), LVM_SETITEMTEXTW, (WPARAM)(i), (LPARAM)&_ms_wlvi); \
+    sfree(_ms_wlvi.pszText);                                            \
+  }
```
Comment (nicholasbishop) [inline src/msapi_utf8.h]: Are these additions used?

File: src/msapi_utf8.h [hunk 6]
```diff
@@ -96,6 +115,81 @@ static __inline wchar_t* utf8_to_wchar(const char* str) {
   return wstr;
 }
 
+/*
+* Converts an non NUL-terminated UTF-16 string of length len to UTF8 (allocate
+* returned string)
+* Returns NULL on error
+*/
+static __inline char* wchar_len_to_utf8(const wchar_t* wstr, int wlen) {
```
Comment (nicholasbishop) [inline src/msapi_utf8.h]: Is this function used?

**Pull request merged**